src/test/java/codeu/controller/RegisterServletTest.java

```diff
@@ -1,35 +1,101 @@
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.mockito.ArgumentCaptor;
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mockito;
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
 
 public class RegisterServletTest {
 
- private RegisterServlet registerServlet;
- private HttpServletRequest mockRequest;
- private HttpServletResponse mockResponse;
- private RequestDispatcher mockRequestDispatcher;
-
- @Before
- public void setup() {
-   registerServlet = new RegisterServlet();
-   mockRequest = Mockito.mock(HttpServletRequest.class);
-   mockResponse = Mockito.mock(HttpServletResponse.class);
-   mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
-   Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/register.jsp"))
-       .thenReturn(mockRequestDispatcher);
- }
-
- @Test
- public void testDoGet() throws IOException, ServletException {
-   registerServlet.doGet(mockRequest, mockResponse);
-
-   Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
- }
+  private RegisterServlet registerServlet;
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+
+  @Before
+  public void setup() {
+    registerServlet = new RegisterServlet();
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/register.jsp")).thenReturn(mockRequestDispatcher);
+  }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+    registerServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoPost_BadUsername() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
+
+    registerServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verifyZeroInteractions(mockResponse);
+    Mockito.verify(mockRequest).setAttribute("error", "Please enter only letters, numbers, and spaces.");
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoPost_NewUser() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("Testertest");
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("password");
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = Mockito.mock(PrintWriter.class);
+    Mockito.when(mockResponse.getWriter()).thenReturn(printWriter);    
+
+    UserStore mockUserStore = Mockito.mock(UserStore.class);
```

Comment:
Since the mockUserStore is used in both this test and the ExistingUser one, you could make it an instance variable and initialize it in setup() (including the setUserStore call).

Same goes for mockSession and its Mockito.when below.